python/nano/src/bigdl/nano/pytorch/torch_nano.py

```diff
@@ -180,12 +243,21 @@ def setup(self, model: nn.Module,    # type: ignore[override]
         # convert single optimizer to a optimizer list
         optimizers = [optimizer] if isinstance(optimizer, Optimizer) else optimizer
 
-        model, optimizers = self._setup(model, optimizers, move_to_device=move_to_device)
+        if self.use_ipex and not TORCH_VERSION_LESS_1_10:
+            (backup_model, backup_optimizers) = copy.deepcopy((model, optimizers))
+            model, optimizers = self._setup(model, optimizers, move_to_device=move_to_device)
+            model_wrapper = _TorchNanoModuleWrapper(model, self, backup_model,  # type: ignore
+                                                    backup_optimizers, move_to_device)
+        else:
+            model, optimizers = self._setup(model, optimizers, move_to_device=move_to_device)
+            model_wrapper = _TorchNanoModuleWrapper(model, self, model,  # type: ignore
+                                                    optimizers, move_to_device)
```

Comment:
1) Why not handle this in `_setup` instead?
2) I think you can extend `_TorchNanoModule` instead of introducing a new class `_TorchNanoModuleWrapper`? `_TorchNanoModule` is already a wrapper class and also deals with `__getattr__`.

Reply:
if we extend `_TorchNanoModule` instead of using `_TorchNanoModuleWrapper`, then the `load_state_dict` method will be:
```python
class TorchNanoModule(_LiteModule):
    def load_state_dict(self, state_dict, xxx):
        backup_model.load_state_dict(state_dict)
        new_model = xxx._setup(backup_model)
        self = new_model
```

however, `self = new_model` is invalid in python, we cann't replace `self` in the `load_state_dict` method

Reply:
> if we extend `_TorchNanoModule` instead of using `_TorchNanoModuleWrapper`, then the `load_state_dict` method will be:
> 
> ```python
> class TorchNanoModule(_LiteModule):
>     def load_state_dict(self, state_dict, ...):
>         backup_model.load_state_dict(state_dict)
>         new_model = xxx._setup(backup_model)
>         self = new_model
> ```
> 
> however, `self = new_model` is invalid in python, we can't replace `self` in the `load_state_dict` method

We can have `_setup`  to return the normal model (possibly optimized by ipex), and wrap it with `_TorchNanoModule` in `setup`; then we can have something like:

```python
class TorchNano(LightningLite):
  def setup(...)
        ...
        model, optimizers = self._setup(model, optimizers, ...)
        model = _TorchNanoModule(model, self._precision_plugin)
        ...

class TorchNanoModule(_LiteModule):
  def load_state_dict(self, state_dict, ...):
        backup.load_state_dict(state_dict)
        module = xxx._setup(backup_model)
        ...
```